usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,40 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+START -> The state of the instance when this code is executed
+         This code is executed by a timer
+
+    START   LICENSE CHANGED TO     CACHE   ACTION
+1   BYOS-1  PAYG                   No      Register - Update Infra
+2   BYOS-1  PAYG                   BYOS    Register - Update Infra
+3   BYOS-1  BYOS-1                 No      Do nothing
+4   BYOS-1  BYOS-1                 BYOS    Do nothing
+5   BYOS-1  BYOS-2                 No      Do nothing
+6   BYOS-1  BYOS-2                 BYOS    Do nothing
+
+7   BYOS-2  PAYG                   No      Do nothing
+8   BYOS-2  PAYG                   BYOS    Re-register - Update Infra
+9   BYOS-2  BYOS-1                 No      De-register - Update Infra
+10  BYOS-2  BYOS-1                 BYOS    De-register - Update Infra
```

Comment:
But the customer cannot switch to an unregistered state via AHB. So the condition expressed here really is

- Customer ran `registercloudguest --clean`
- Customer used AHB to explicitly set the BYOS license

As such there is nothing to do. If the starting state is BYOS, any form not registered or registered to the update infra or SCC and the cache says BYOS we have nothing to do. If there is no cache we also do nothing because we cannot do anyting.